beacon_node/beacon_chain/src/fetch_blobs.rs

```diff
@@ -0,0 +1,305 @@
+                    // publish components already seen on gossip.
+                    let is_supernode = chain_cloned
+                        .data_availability_checker
+                        .get_custody_columns_count()
+                        == spec.number_of_columns;
+
+                    // At the moment non supernodes are not required to publish any columns.
+                    // TODO(das): we could experiment with having full nodes publish their custodied
+                    // columns here.
+                    if !is_supernode {
+                        return;
+                    }
+
+                    // To reduce bandwidth for supernodes: permute the columns to publish and
+                    // publish them in batches. Our hope is that some columns arrive from
+                    // other supernodes in the meantime, obviating the need for us to publish
+                    // them. If no other publisher exists for a column, it will eventually get
+                    // published here.
+                    // FIXME(das): deduplicate this wrt to gossip/sync methods
+                    all_data_columns.shuffle(&mut rand::thread_rng());
+
+                    let batch_size =
+                        all_data_columns.len() / SUPERNODE_DATA_COLUMN_PUBLICATION_BATCHES;
+                    for batch in all_data_columns.chunks(batch_size) {
+                        let already_seen = chain_cloned
+                            .data_availability_checker
+                            .imported_custody_column_indexes(&block_root)
+                            .unwrap_or_default();
+                        let publishable = batch
+                            .iter()
+                            .filter(|col| !already_seen.contains(&col.index()))
+                            .cloned()
+                            .collect::<Vec<_>>();
+                        if !publishable.is_empty() {
+                            debug!(
+                                chain_cloned.log,
+                                "Publishing data columns from EL";
+                                "count" => publishable.len()
+                            );
+                            publish_fn(BlobsOrDataColumns::DataColumns(publishable));
+                        }
+                        tokio::time::sleep(SUPERNODE_DATA_COLUMN_PUBLICATION_BATCH_INTERVAL).await;
+                    }
+                },
+                "compute_data_columns",
+            )
+            .ok_or(FetchEngineBlobError::RuntimeShutdown)?;
+
+        Some(data_columns_receiver)
+    } else {
+        if num_fetched_blobs == 0 {
+            debug!(
+                chain.log,
+                "No blobs fetched from the EL";
+                "num_expected_blobs" => num_expected_blobs,
+            );
+            inc_counter(&metrics::BLOBS_FROM_EL_MISS_TOTAL);
+            return Ok(None);
+        }
+
+        inc_counter(&metrics::BLOBS_FROM_EL_HIT_TOTAL);
```

Comment:
Again, very different meaning to the metric in peerdas

Reply:
Addressed in 0a9d5a0f61b4114dff93fe88c46ca4b97dde75f2